src/riak_cs_console.erl

```diff
@@ -100,10 +102,104 @@ cleanup_orphan_multipart(Timestamp) when is_binary(Timestamp) ->
+    io:format("~p: getting ~p:~p~n", [erlang:now(), RawBucket, RawKey]),
+    case riakc_pb_socket:get(Pid, RawBucket, RawKey, GetOptions, GetTimeout) of
+        {ok, RiakObj} ->
+            io:format("~p: Trying to resolve ~p sibling(s) of ~p:~p~n",
+                      [erlang:now(), riakc_obj:value_count(RiakObj), RawBucket, RawKey]),
+            _ = lager:info("Trying to resolve ~p sibling(s) of ~p:~p",
+                           [riakc_obj:value_count(RiakObj), RawBucket, RawKey]),
+            case resolve_ro_siblings(RiakObj, RawBucket, RawKey) of
+                {ok, RiakObj2} ->
+                    R = riakc_pb_socket:put(Pid, RiakObj2, PutOptions, PutTimeout),
+                    io:format("~p: Resolution result: ~p~n", [erlang:now(), R]);
+                {error, R} = E ->
+                    io:format("~p: Not updating ~p:~p: ~p~n",
+                              [erlang:now(), RawBucket, RawKey, R]),
+                    E
+            end;
+        {error, Reason} = E ->
+            io:format("~p: Failed to get an object before resolution: ~p~n",
+                      [erlang:now(), Reason]),
+            E
+    end.
+
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
 
+-spec resolve_ro_siblings(riakc_obj:riakc_obj(), binary(), binary()) ->
+                                 {ok, riakc_obj:riakc_obj()} | {error, term()}.
+resolve_ro_siblings(_, ?USER_BUCKET, _) ->
+    %% This is not supposed to happen unless something is messed up
+    %% with Stanchion. Resolve logic is not obvious and needs operator decision.
+    {error, {not_supported, ?USER_BUCKET}};
+resolve_ro_siblings(_, ?ACCESS_BUCKET, _) ->
+    %% To support access bucket, JSON'ized data should change its data
+    %% structure to G-set, or map(register -> P-counter)
+    {error, {not_supported, ?ACCESS_BUCKET}};
+resolve_ro_siblings(_, ?STORAGE_BUCKET, _) ->
+    %% Storage bucket access conflict resolution is not obvious; maybe
+    %% adopt max usage on each buckets.
+    {error, {not_supported, ?STORAGE_BUCKET}};
+resolve_ro_siblings(_, ?BUCKETS_BUCKET, _) ->
+    %% Bucket conflict resolution obvious, referring to all users
+    %% described in value and find out who's true user. If multiple
+    %% users are owner (that cannot happen unless Stanchion is messed
+    %% up), operator must choose who is true owner.
+    {error, {not_supported, ?BUCKETS_BUCKET}};
+resolve_ro_siblings(RO, ?GC_BUCKET, _) ->
+    Resolved = riak_cs_gc:decode_and_merge_siblings(RO, twop_set:new()),
+    Obj = riak_cs_utils:update_obj_value(RO,
+                                         riak_cs_utils:encode_term(Resolved)),
+    {ok, Obj};
+resolve_ro_siblings(RO, <<"0b:", _/binary>>, _) ->
+    case riak_cs_utils:resolve_robj_siblings(riakc_obj:get_contents(RO)) of
+        {{MD, Value}, true} when is_binary(Value) ->
```

Comment:
Wondering how tombstones should be treated... Put them back without deleting or delete them explicitly.